test/unit/model/Service/TimeoutServiceTest.php

```
@@ -210,9 +233,14 @@ private function expectTestMap(array $testMap): void
             ->willReturn($testMap);
     }
 
-    private function createCommand(bool $hastStartTimer = false, bool $lateSubmissionAllowed = false): TimeoutCommand
+    private function createCommand(bool $hastStartTimer = false, bool $lateSubmissionAllowed = false, bool $permanentLateSubmission =  false): TimeoutCommand
```

PHPMD (2.15.0)

The method createCommand has a boolean flag argument $hastStartTimer, which is a certain sign of a Single Responsibility Principle violation. (Clean Code Rules)

(BooleanArgumentFlag)